Sources/PersistentCollections/Node/_Node+isEqual.swift

```diff
@@ -10,40 +10,46 @@
 //===----------------------------------------------------------------------===//
 
 // TODO: `Equatable` needs more test coverage, apart from hash-collision smoke test
-extension _Node: Equatable where Value: Equatable {
+extension _Node {
   @inlinable
-  static func == (left: _Node, right: _Node) -> Bool {
-    if left.raw.storage === right.raw.storage { return true }
+  internal func isEqual(
+    to other: _Node,
+    by areEquivalent: (Element, Element) -> Bool
+  ) -> Bool {
```

Comment:
The cost is usually pretty minimal with these -- this is a non-escaping closure, so it can and will be stack-allocated when possible! Inlining of the closure may trigger as well, but I don't really have an intuition whether it is likely to happen here (tbh I don't even know if the compiler would be willing to specialize functions for specific closures like that).

If I had some free time, I would experiment with switching `_Node.Element` from a tuple to an internal frozen struct, so it would be able to carry an (easily inlinable) Equatable conformance.

Reply:
On second look, it looks I ended up changing `isSubset(of:)` and `isDisjoint(with:)` to get rid of the closure -- it superficially allows degrees of freedom that aren't actually supported. (I.e., we cannot really customize key comparisons, as that would conflict with our hashing scheme.) I'll replace it with a generic argument for Value instead.